powdr_cli/src/main.rs

```diff
@@ -318,6 +319,7 @@ fn main() {
                 ));
             }
         }
+        #[cfg(feature = "halo2")]
         Commands::Prove {
```

Comment:
I don't think this should be enabled only for halo2

Reply:
I agree with you, but unless we have another backend, this path is unreachable, and there is no good generalization for it.

The issue is that variable `backend` can not be constructed, because the enum is empty with halo2 disabled. Then inside the function `read_and_prove`, called from this match arm, there is an exhaustive match with all possible backends to generate the function output, which I can't do if there is no backend. In fact, it renders this whole code path unreachable, because it assumes there is a backend, but there can't be one.

I think this is the best solution given the situation, and leave this this to be fixed when we have other backends, because then we will probably have to generalize how we use the different backends with a single API (see my TODO below).

Reply:
right ok